feat: 계획블록 생성

m1njae opened the pull request:
<!-- 
- PR 제목: [feat/fix/refactor...] 작업 내용 한 줄 요약 (브랜치 이름)
- commit message가 적절한지 확인해주세요.
- 적절한 branch로 요청했는지 확인해주세요.
- Assignees, Label을 붙여주세요.
- 주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 Reviewer로 등록해주세요.
- PR이 승인된 경우 해당 브랜치는 삭제 부탁드립니다. 
-->


## 📝 PR 요약

<!--
해당 pr에서 작업한 내역을 적어주세요.
-->

- 일간, 루틴로드 계획 블록을 생성합니다.

<br/>

## 📌 변경 사항 및 주의 사항

<!--
변경사항 및 주의 사항이 있다면 적어주세요.
주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 리뷰어로 등록해주세요. (다른 사람이 작성한 코드 수정 등)
코드 리뷰 시 더 꼼꼼하게 확인 받고 싶은 부분이 있다면 적어주세요.
-->
- ~dayjs를 추가하였습니다. 나중에 pull 받고 yarn 실행해주세요!~
- 조회랑 생성에서 보내주는 planDate를 보내주는 타입이 다른거 같아서 이 부분을 서팟과 논의해보고 싶습니다. 계속 고민이 되네요!
- 하루 계획 블록을 처음 생성할 때, 그날의 planOrder도 생성해주어야 할 것 같아서 그 부분까지 고려해서 구현해보았습니다.
- 잘 짜보겠다고 짜긴 했는데.. 제법 복잡하게 되었네요.. 제가 생각한 로직 구현에 대한 방향은 주석을 달아놓았으니, 참고해주시면 될 것 같습니다!


![image](https://user-images.githubusercontent.com/82744423/213991835-e6c2b366-2c6d-4d70-a131-500667212fe4.png)

<br/>

## 🔗 Linked Issue
close #22

m1njae committed:
add: dayjs 설치

m1njae committed:
chore: responseMessage 추가

m1njae committed:
chore: planDate 타입 변경

m1njae committed:
feat: 계획블록 생성

m1njae committed:
fix: type이 routine일 때 로직 수정

Chedda98 commented on src/controllers/PlanController.ts:
```diff
@@ -61,4 +65,44 @@ export class DailyPlanController {
+    @Res() res: Response,
+    @Param('userId') userId: string,
+    @BodyParam('planName') planName: string,
+    @BodyParam('date') date: string,
```
@m1njae postman 테스트 해본 캡쳐 화면이랑 코드를 확인해보고 궁금한 점이 생겨서요!!  
클라쪽에서 date를 보내주면 우리쪽에서 planDate로 출력해주는 것이 맞나요?!

m1njae replied:
맞습니다!

Chedda98 commented on src/services/PlanService.ts:
```diff
@@ -44,4 +46,151 @@ export class DailyPlanService {
+            where: {
+              user_id: userId,
+              type: type,
+              planDate: date,
```
해당 부분이 위에 질문과 이어지는 곳입니다! 
클라에게 body로 param값 받을 때도 `planDate`로 받으면 따로 값 변환 없이 바로 그 값을 data에 넣어도 되지 않을까 싶어서요!!

m1njae replied:
음 그렇네요!  type과 date를 쿼리로 받을 때 type, date으로 받았던터라, body에서도 같은 값으로 받으면 혼동이 덜 하지 않을까 생각했었습니다! 아니면 아예 쿼리로 받을 때도 `planDate`로 받을까요?! 체다는 어떻게 생각하세요?!🙌

Chedda98 commented on src/services/PlanService.ts:
```diff
@@ -44,4 +46,151 @@ export class DailyPlanService {
+          });
+
+          if (existPlanOrder.length == 0) {
+            //* planOrder가 없다면 그 날의 daily, reschedule planOrder 생성
```
제가 `planOrder`쪽이 잘 이해가 되지 않아서 이쪽 나중에 게더에서 만나서 설명해주실 수 있나요?!!🥹🥹🥹 @m1njae

m1njae replied:
넵넵 물론입니당 논리적인 글 주변이 없어서😢 흑흑 게더에서 구두로 말씀해드릴게요!

m1njae committed:
chore: merge 이후 코드 정리

m1njae committed:
chore: errorValidator 주석 처리

body 값을 유효성 검사하는 과정에서 계속해서 errorValidator가 동작하지 않습니다.
저의 개인적인 생각은 class-transform 하는 과정 때문에 동작하지 않는 것 같은데,
당장은 시간이 부족하니 에러 핸들링을 if 조건문으로 구현하였습니다.

m1njae committed:
chore: planController 코드 정리

m1njae committed:
chore: planService 로직 수정

루틴 로드의 경우 우회할 계획과 마찬가지로 날이 바뀌어도 유지가 되어야 하므로
시간 데이터를 저장하지 않습니다.
